src/EventListener/WriteListener.php

```diff
@@ -62,12 +63,15 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
 
                 $event->setControllerResult($persistResult ?? $controllerResult);
 
-                // Controller result must be immutable for _api_write_item_iri
```

Comment:
This change should be reverted isn't it? At least we need to keep the comment.

Reply:
i don't know what that line means specifically, but I don't see why we should not generate and IRI for any controller result.

without this change, these lines will fail:
https://github.com/api-platform/core/pull/2482/files#diff-471e9b113802e057f7f37436181d4b55R121

https://github.com/api-platform/core/pull/2482/files#diff-471e9b113802e057f7f37436181d4b55R152

Reply:
Can you take a look @soyuka? I'm not confident at all in this part of the code base, we may introduce BC breaks...

Reply:
specifically, the `resource_class` here is the abstract class, because we are hitting the operation defined on the abstract class, but the controller result will be the concrete implementation.

Reply:
it sets the Content-Location in the RespondListener

Reply:
Or would be more solid IMP

Reply:
ok, looks not possible. in the `RespondListener`, the controller response is serialized (run after `SerializeListener`), so i have nothing to generate an IRI from. So, i think that's why it's done here in the `WriteListener`